### PR TITLE
Site header styling

### PR DESCRIPTION
**Pull request by DJMyles**
I'm not that well versed on best practice for SVG so it's likely this is the wrong approach. 

Two parts to this branch:

- the replacement image for the logo to SVG along with the preprocessing to get it to pick it up
- the styles that override links in the logo section of the header

### CODE REVIEW

**Comment by Feng-Shui** [comment]
I've just pushed another commit to this branch, it drops the logo in via CSS.  We're going to need to sort out how to deal with (ideally SVG) logos of different sizes if the base theme is going to be run without customisation.  I've got a few thoughts re theme settings and such, but that's out of scope for the MVP so I think we should park that for now.  @DJMyles take a look at the code, it's pretty much inline with the UI Kit version now I think.

**Comment by Feng-Shui** [comment]
This will also need some browser testing to make sure the svg background image plays nice.  Looking at you, IE... But we can target those browsers specifically and get them to fallback on the png if needed.

**Comment by rujiali** [comment]
Question: we don't support user uploaded logo ? @Feng-Shui

**Comment by Feng-Shui** [comment]
Yeah, we should... leave that with me.

**Comment by Feng-Shui** [comment]
Ok...

Drupal 7 doesn't support svg out of the box, which kinda sucks, but I still want to use an SVG logo in the theme, BUT, we need to allow people to upload custom logos and have them just work.... so:

1. The svg is only validated by Drupal on upload, if you specify a path to a custom logo in the theme settings then you can pick whatever you want, so I've provided some default theme settings that disable the default icon and point to the svg version instead.

1. An end user can chose to upload their own logo, and that simply overwrite the default logo path that we provide.

1. I've provided a new theme setting for a maximum logo height (default 150px), this has two purposes:
    1. It gives a non-technical user who doesn't have the ability to resize images the ability to put a logo up and not have it break the header layout.
    1. It also provides a height attribute for the svg (in `template.php`) which means that the svg is constrained (the browser will do the width by default) and will render ok - this can then be overridden at any breakpoint if you want to do so in your subtheme.

1. We've still got a png version in the theme as a backup.

Let me know what you guys think.

@rujiali we also need to provide a default option in the info file for the sidebar setting you have in the form.

**Comment by DJMyles** [comment]
@Feng-Shui I'm still not seeing the logo, am I missing something? it's enabled in the theme settings.

File: templates/page.tpl.php [hunk 1]
```diff
@@ -18,13 +18,9 @@
     <div class="wrapper">
 
       <div class="page-header__logo">
-        <?php if ($logo): ?>
-          <a href="<?php print $front_page; ?>" title="<?php print t('Home'); ?>" rel="home" class="header__logo logo" id="logo"><img src="<?php print $logo; ?>" alt="<?php print t('Home'); ?>" class="header__logo-image" /></a>
-        <?php endif; ?>
-        <a href="<?php print $front_page; ?>" title="<?php print t('Home'); ?>" rel="home" class="header__logo logo" id="logo"><?php print $site_name; ?></a>
```
Comment (rujiali) [inline templates/page.tpl.php:36]: Do we remove the site-name too ? @Feng-Shui

Comment (Feng-Shui) [reply templates/page.tpl.php:36]: Nope, I've just dropped that back in, with the ability to turn on/off site name and slogan.  Although these currently appear below the logo.

@DJMyles can you take a look at the styling here?  We need to site name and slogan to be set to a decent font size and then to sit beside the logo.

**Comment by Feng-Shui** [comment]
I've made some final tweaks and changes to this branch.  These were mainly to get the vertical alignment of elements working to support different logo sizes.  I've also made some changes to make sure things like a header search bar doesn't wrap on tablet.

All of the header search form styling has been changed from targeting `<form>` elements (which may clash with other content in the region) to `.search-form`.

@rujiali @DJMyles can you guys please take this for a test - would like to get this merged soon :)

**Comment by DJMyles** [comment]
@Feng-Shui no issues apart from the grey border extending across to the button as well. Happy for this to be merged and I'll tackle that as a separate issue.

**Comment by Feng-Shui** [comment]
@DJMyles cool, yeah please open a new issue for that :)

**Pull request merged**